contrib/babelfishpg_tsql/src/catalog.c

```diff
@@ -2576,7 +2580,7 @@ rename_procfunc_update_bbf_catalog(RenameStmt *stmt)
 	funcsign_datum = heap_getattr(usertuple, Anum_bbf_function_ext_funcsignature,
 								  bbf_func_ext_rel->rd_att, &is_null);
 	funcsign = pstrdup(TextDatumGetCString(funcsign_datum));
-	new_funcsign = strcat(pstrdup(stmt->newname), strrchr(funcsign, '('));
+	snprintf(new_funcsign, sizeof(new_funcsign), "%s%s", stmt->newname, strrchr(funcsign, '('));
 
 	new_record_func_ext[Anum_bbf_function_ext_funcname - 1] = CStringGetDatum(stmt->newname);
 	new_record_func_ext[Anum_bbf_function_ext_orig_name - 1] = CStringGetTextDatum(orig_proc_funcname);
```

Comment:
Can `orig_proc_funcname` be NULL? If not, please add an `Assert(orig_proc_funcname)` here because we're using some global variable that this function never set

Reply:
If orig_proc_funcname is NULL, I think we don't need to update this column. Example: I remember the function created for trigger doesn't have a orig_funcname